File: devops/java/platform_check.java
public class PlatformDetector {

    public static void main(String[] args) {
        // Get the operating system name
        String osName = System.getProperty("os.name");
        // Get the operating system architecture
        String osArch = System.getProperty("os.arch");
        // Get the operating system version
        String osVersion = System.getProperty("os.version");

        // Print the detected platform information
        System.out.println("Operating System Name: " + osName);
        System.out.println("Operating System Architecture: " + osArch);
        System.out.println("Operating System Version: " + osVersion);

        // Determine the type of operating system
        if (osName.toLowerCase().contains("win")) {
            System.out.println("Platform: Windows");
        } else if (osName.toLowerCase().contains("mac")) {
            System.out.println("Platform: Mac OS");
        } else if (osName.toLowerCase().contains("nix") || osName.toLowerCase().contains("nux") || osName.toLowerCase().contains("aix")) {
            System.out.println("Platform: Unix/Linux");
        } else {
            System.out.println("Platform: Unknown");
        }
    }
}

File: devops/perlbook/platform_check.pl
use strict;
use warnings;
use Config;
use POSIX qw(uname);

# Get the operating system name using Config module
my $os_name = $Config{osname};

# Get the operating system version using Config module
my $os_version = $Config{osvers};

# Get the operating system name using POSIX module
my $posix_os_name = uname();

# Print the detected platform information
print "Operating System Name (Config): $os_name\n";
print "Operating System Version (Config): $os_version\n";
print "Operating System Name (POSIX): $posix_os_name\n";

# Determine the type of operating system
if ($os_name =~ /win/i) {
    print "Platform: Windows\n";
} elsif ($os_name =~ /darwin/i) {
    print "Platform: Mac OS\n";
} elsif ($os_name =~ /linux|unix|aix/i) {
    print "Platform: Unix/Linux\n";
} else {
    print "Platform: Unknown\n";
}
